pkgs/development/python-modules/spacy/default.nix

```
@@ -53,6 +53,10 @@ buildPythonPackage rec {
   #   ${python.interpreter} -m pytest spacy/tests --vectors --models --slow
   # '';
 
+  postPatch = ''
+    substituteInPlace setup.cfg --replace "thinc==7.4.0" "thinc>=7.4.0,<7.5.0"
```

We should take up to the next major bump, minor bumps shouldn't introduce breaking changes. Also we will have to revisit this again if they do another minor bump
```suggestion
    substituteInPlace setup.cfg --replace "thinc==7.4.0" "thinc>=7.4.0,<8"
```

---

I was a bit hesitant, since IIRC they broke something before in a minor bump (which is probably why they set a hard constraint). But since they are working on thinc 0.8 now, it's not very likely to break now.

It seems that GH's suggest functionality cannot amend an existing commit, so I have just manually updated with your suggestion.

---

> I was a bit hesitant, since IIRC they broke something before in a minor bump (which is probably why they set a hard constraint). But since they are working on thinc 0.8 now, it's not very likely to break now.

This is why tests are important :), so we should see breakages if thinc is updated.

Either way, nix doesn't follow conventional python version bound guidelines anyway.